packages/fuels/src/setup-launch-node-server.test.ts

```diff
@@ -0,0 +1,93 @@
+import { Provider } from '@fuel-ts/account';
+import { waitUntilUnreachable } from '@fuel-ts/utils/test-utils';
+import { spawn } from 'node:child_process';
+
+function startServer(
+  port: number = 0
+): Promise<{ serverUrl: string; killServer: () => void } & Disposable> {
+  return new Promise((resolve, reject) => {
+    const cp = spawn(`pnpm tsx packages/fuels/src/setup-launch-node-server.ts ${port}`, {
+      detached: true,
+      shell: 'sh',
+    });
+
+    const killServer = () => {
+      // https://github.com/nodejs/node/issues/2098#issuecomment-169549789
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      process.kill(-cp.pid!);
+    };
+
+    cp.stdout?.on('data', (chunk) => {
+      // first message is server url
+      const message: string[] = chunk.toString().split('\n');
+      const serverUrl = message[0].startsWith('http://') ? message[0] : '';
+      // teardown
+      resolve({
+        serverUrl,
+        killServer,
+        [Symbol.dispose]: killServer,
+      });
```

Comment:
we should probably handle any potential errors that could arise with the `serverUrl`

```suggestion
    if (serverUrl) {
      resolve({
        serverUrl,
        killServer,
        [Symbol.dispose]: killServer,
      });
    } else {
      reject(new Error('Invalid server URL'));
    }
  ```

Reply:
I simplified the logic a bit [here](https://github.com/FuelLabs/fuels-ts/pull/2378/commits/7a1ebbfe1a2944f5aebf3de9e75bd6003ba8b497#diff-fdddb3f57680c5f4a5d071b37425ff8f81fd98207894622482e7e9a637f2e7e0R28). Keep in mind that this is just a test helper and the tests will fail if the `serverUrl` is invalid in any way.